tests/OnlineSales.Tests/SimpleTableTests.cs

```diff
@@ -167,6 +167,42 @@ public async Task InvalidLimit(string filter, int dataCount)
         await GetTest($"{this.itemsUrl}?{filter}", HttpStatusCode.BadRequest);
     }
 
+    [Theory]
+    [InlineData("filtercadabra", HttpStatusCode.BadRequest)]
+    [InlineData("filter", HttpStatusCode.BadRequest)]
+    [InlineData("filter[]", HttpStatusCode.BadRequest)]
+    [InlineData("filter[]=", HttpStatusCode.BadRequest)]
+    [InlineData("filter[]=0", HttpStatusCode.BadRequest)]
+    [InlineData("filter[][]=3", HttpStatusCode.BadRequest)]
+    [InlineData("filter[][][]=4", HttpStatusCode.BadRequest)]
+    [InlineData("filter[notexists]=5", HttpStatusCode.BadRequest)]
+    [InlineData("filter[where][notexists]=6", HttpStatusCode.BadRequest)]
+    [InlineData("filter[where][][]=7", HttpStatusCode.BadRequest)]
+    [InlineData("filter[where][id][]=8", HttpStatusCode.BadRequest)]
+    [InlineData("filter[where][id][notexists]=9", HttpStatusCode.BadRequest)]
+    [InlineData("filter[^7@5\\nwhere][id^7@5\\n][|^7@5\\n]=^7@5\\n", HttpStatusCode.BadRequest)]
+    [InlineData("filter[where][id]=^7@5\\n", HttpStatusCode.BadRequest)]
+    [InlineData("filter[][id]=^7@5\\n", HttpStatusCode.BadRequest)]
+    [InlineData("filter[where][id][eq]=^7@5\\n", HttpStatusCode.BadRequest)]
+    [InlineData("filter[limit]=abc", HttpStatusCode.BadRequest)]
+    [InlineData("filter[skip]=abc", HttpStatusCode.BadRequest)]
+    [InlineData("filter[order]=5555incorrectfield777", HttpStatusCode.BadRequest)]
+    public async Task InvalidQueryParameter(string filter, HttpStatusCode code)
+    {
+        await GetTest($"{this.itemsUrl}?{filter}", code);
+    }
+
+    [Theory]
+    [InlineData("filter[where][id]=9", HttpStatusCode.OK)]
```

Comment:
@Exmirai I can see no test which verify how different attribute types like bool, string, date and different  operands (gt, ls, gte, etc) work.